### PR TITLE
Parallel Plate Capacitor and Solenoid Inductance

### PR DESCRIPTION
**Pull request by miguelzaqui**
Since previous versions, there is a function to calculate the **resistance of a wire** (`wireresistance`). As a complement, the functions for calculating the **capacitance of a parallel plate capacitor** (`parallel_plate_capacitance`) and the **inductance of a solenoid** (`solenoid_inductance`) have been added.

### CODE REVIEW

**Comment by Lakshmikanth2001** [comment]
Thank you very much... , Please try to include `test cases` for your functions so that we can test your function across various input 

please review `test` directory of our project

File: electricpy/__init__.py [hunk 1]
```diff
@@ -4782,6 +4782,102 @@ def wireresistance(length=None,diameter=None,rho=16.8*10**-9,R=None):
...
         return _np.sqrt(4*A/pi)
+
+def parallel_plate_capacitance(A=None, d=None, e=e0, C=None):
+    r"""
+    Parallel-Plate Capacitance Calculator.
+
+    Enter three values to calculate the remaing one. Even though every variable
+    is unitless, please use the International System of Units.
+
+    .. math:: C =\frac{\varepsilon A}{d}
```
Comment (engineerjoe440) [inline electricpy/__init__.py:4793]: consider using `\cdot` to denote multiplication in the numerator, it makes things explicitly clear that there's multiplication taking place

File: electricpy/__init__.py [hunk 2]
```diff
@@ -4782,6 +4782,102 @@ def wireresistance(length=None,diameter=None,rho=16.8*10**-9,R=None):
...
+        return e*A/C
+
+def solenoid_inductance(A=None, l=None, N=None, u=u0, L=None):
+    r"""
+    Solenoid Inductance Calculator.
+
+    Enter four values to calculate the remaing one. Even though every variable
+    is unitless, please use the International System of Units.
+
+    .. math:: L = \frac{\mu N^2 A}{l}
```
Comment (engineerjoe440) [inline electricpy/__init__.py:4837]: consider using `\cdot` to denote multiplication in the numerator, it makes things explicitly clear that there's multiplication taking place

File: electricpy/__init__.py [hunk 3]
```diff
@@ -4782,6 +4782,102 @@ def wireresistance(length=None,diameter=None,rho=16.8*10**-9,R=None):
...
+    Parallel-Plate Capacitance Calculator.
+
+    Enter three values to calculate the remaing one. Even though every variable
+    is unitless, please use the International System of Units.
+
+    .. math:: C =\frac{\varepsilon A}{d}
+
+    Parameters
+    ----------
+    A:  [float]
```
Comment (engineerjoe440) [inline electricpy/__init__.py:4797]: The square brackets are not explicitly needed to denote parameters of a single type (float, in this case). Although it's fine to use them, this practice would seem inconsistent with the rest of the library.

- [ ] remove square brackets from parameter type-strings

**Review by engineerjoe440 — CHANGES_REQUESTED** [review]

**Comment by engineerjoe440** [comment]
I wanted to say ***thank you*** for your contributions! I greatly appreciate your interest and support. I think that there is great value in the functions which you've added. I had a few minor suggestions mostly around general formatting. I can't wait to see these added! :)

I want to say thank you, also, to @Lakshmikanth2001. He has been incredibly helpful, and has kept the project moving when I've been terribly slow. His points are valid about adding test cases. Little tests will make a big difference and will help make sure that folks like me don't accidentally break your functions as maintenance continues.

File: electricpy/__init__.py (hunk 3 above)
Comment (miguelzaqui) [reply electricpy/__init__.py:4797]: It's a pleasure to collaborate! I like the changes you propose. I was relying on the `wireresistance` function to write the documentation for `parallel_plate_capacitance` and `solenoid_inductance`. However, I noticed that in the `wireresistance` function, there are parameters indicated as: `[float], optional`. That is, they are also indicated with brackets. Additionally, I just realized that I forgot to put that the parameters are optional. I then propose to place `float, optional`.

**Comment by miguelzaqui** [comment]
> Thank you very much... , Please try to include `test cases` for your functions so that we can test your function across various input
> 
> please review `test` directory of our project

I was just implementing the tests of these functions. I confess that I am not very familiar with the subject of tests in Python, but you can review it and tell me if it meets the requirements.

File: electricpy/__init__.py (hunk 3 above)
Comment (engineerjoe440) [reply electricpy/__init__.py]: That looks fantastic! You know, you're probably right! I bet the `wireresistance` function should be updated to change that too. We typically use the `[]` brackets when there are two or more types that could be used; something like: `[str, float]`

Thanks for tackling this! :)

File: electricpy/__init__.py [hunk 4]
```diff
@@ -4782,6 +4782,102 @@ def wireresistance(length=None,diameter=None,rho=16.8*10**-9,R=None):
...
+    A:  float, optional
+        Area of the plate, unitless.
+    d:  float, optional
+        Distance between the plates, unitless.
+    e:  float, optional
+        Permitivity of the dielectric, unitless.
+        Default value is the permittivity of free space: 8.854E-12
+    C:  float, optional
+        Capacitance, unitless.
+    Returns
```
Comment (engineerjoe440) [inline electricpy/__init__.py:4806]: pydocstyle pointed out an inconsistency here; we should have an extra blank line prior to the "Return" section as highlighted by these errors:

```
./__init__.py:4787 in public function `parallel_plate_capacitance`:
        D4[10](https://github.com/engineerjoe440/ElectricPy/runs/5686294364?check_suite_focus=true#step:5:10): Missing blank line after section ('Parameters')
./__init__.py:4787 in public function `parallel_plate_capacitance`:
        D4[11](https://github.com/engineerjoe440/ElectricPy/runs/5686294364?check_suite_focus=true#step:5:11): Missing blank line before section ('Returns')
Error: Process completed with exit code 1.
```

**Review by engineerjoe440 — CHANGES_REQUESTED** [review]
Looks like all we need is a little update to the docstring. Thanks again for this work!

**Review by engineerjoe440 — APPROVED** [review]
Thank you so much for your work!

**Pull request merged**